go/multipooler/manager/pg_replication.go

```diff
@@ -71,129 +71,144 @@ func formatStandbyList(standbyIDs []*clustermetadatapb.ID) string {
 	if err != nil {
 		return false, mterrors.Wrap(err, "failed to check if replay LSN reached target")
 	}
+	var reachedTarget bool
+	if err := executor.ScanSingleRow(result, &reachedTarget); err != nil {
+		return false, mterrors.Wrap(err, "failed to scan LSN comparison result")
+	}
 	return reachedTarget, nil
 }
 
 // queryReplicationStatus queries PostgreSQL for all replication status fields.
 // This method handles NULL values properly for LSN fields that may be NULL
 // when not in recovery mode or when no WAL has been received/replayed.
 func (pm *MultiPoolerManager) queryReplicationStatus(ctx context.Context) (*multipoolermanagerdatapb.StandbyReplicationStatus, error) {
-	var replayLsn sql.NullString
-	var receiveLsn sql.NullString
-	var isPaused bool
-	var pauseState string
-	var lastXactTime sql.NullString
-	var primaryConnInfo string
-
-	query := `SELECT
-		pg_last_wal_replay_lsn(),
-		pg_last_wal_receive_lsn(),
+	sql := `SELECT
+		pg_last_wal_replay_lsn()::text,
+		pg_last_wal_receive_lsn()::text,
 		pg_is_wal_replay_paused(),
 		pg_get_wal_replay_pause_state(),
-		pg_last_xact_replay_timestamp(),
+		pg_last_xact_replay_timestamp()::text,
 		current_setting('primary_conninfo')`
 
 	queryCtx, cancel := context.WithTimeout(ctx, 500*time.Millisecond)
 	defer cancel()
-
-	err := pm.db.QueryRowContext(queryCtx, query).Scan(
-		&replayLsn,
-		&receiveLsn,
-		&isPaused,
-		&pauseState,
-		&lastXactTime,
-		&primaryConnInfo,
-	)
+	result, err := pm.query(queryCtx, sql)
 	if err != nil {
 		return nil, mterrors.Wrap(err, "failed to query replication status")
 	}
 
-	status := &multipoolermanagerdatapb.StandbyReplicationStatus{
-		IsWalReplayPaused:   isPaused,
-		WalReplayPauseState: pauseState,
+	if result == nil || len(result.Rows) == 0 {
+		return nil, mterrors.Wrap(fmt.Errorf("no rows returned"), "failed to query replication status")
 	}
 
-	if replayLsn.Valid {
-		status.LastReplayLsn = replayLsn.String
+	row := result.Rows[0]
+	status := &multipoolermanagerdatapb.StandbyReplicationStatus{}
+
+	// Get replay LSN (may be NULL)
+	if replayLsn, err := executor.GetString(row, 0); err == nil && replayLsn != "" {
+		status.LastReplayLsn = replayLsn
 	}
-	if receiveLsn.Valid {
-		status.LastReceiveLsn = receiveLsn.String
+
+	// Get receive LSN (may be NULL)
+	if receiveLsn, err := executor.GetString(row, 1); err == nil && receiveLsn != "" {
+		status.LastReceiveLsn = receiveLsn
 	}
-	if lastXactTime.Valid {
-		status.LastXactReplayTimestamp = lastXactTime.String
+
+	// Get is_paused (boolean)
+	if isPaused, err := executor.GetBool(row, 2); err == nil {
+		status.IsWalReplayPaused = isPaused
 	}
 
+	// Get pause state (string)
+	if pauseState, err := executor.GetString(row, 3); err == nil {
+		status.WalReplayPauseState = pauseState
+	}
+
+	// Get last xact time (may be NULL)
+	if lastXactTime, err := executor.GetString(row, 4); err == nil && lastXactTime != "" {
+		status.LastXactReplayTimestamp = lastXactTime
+	}
```

Comment:
Similar to LSNs above: NULL and empty string are not the same.
`::text` preserves NULL-ness, it does not convert NULL to empty string.

Reply:
Fixed